src/bio/bio_device.c

```diff
@@ -1012,21 +1173,33 @@ bio_led_manage(struct bio_xs_context *xs_ctxt, char *tr_addr, uuid_t dev_uuid, u
 	       unsigned int *state, uint64_t duration)
 {
 	struct spdk_pci_addr	pci_addr;
+	int                     addr_len = 0;
 	int			rc;
 
+	/* LED management on NVMe devices currently only supported when VMD is enabled. */
+	if (!bio_vmd_enabled)
+		return 0;
+
 	/**
 	 * If tr_addr is already provided, convert to a PCI address. If tr_addr is NULL or empty,
 	 * derive PCI address from the provided UUID and if tr_addr is an empty string buffer then
 	 * populate with the derived address.
 	 */
 
-	if ((tr_addr == NULL) || (strlen(tr_addr) == 0)) {
+	if (tr_addr != NULL) {
+		addr_len = strnlen(tr_addr, SPDK_NVMF_TRADDR_MAX_LEN);
```

Comment:
same as above, if SPDK_NVMF_TRADDR_MAX_LEN is valid, it is better to be:
addr_len = strnlen(tr_addr, SPDK_NVMF_TRADDR_MAX_LEN + 1)

Reply:
done